letta/llm_api/litellm.py

```diff
@@ -0,0 +1,192 @@
+import json
+import os
+from typing import Optional
+
+import requests
+
+from letta.log import get_logger
+from letta.utils import smart_urljoin
+
+logger = get_logger(__name__)
+
+
+def litellm_get_models_prices_and_context_window() -> dict[str, dict]:
+    """Reads model prices and context window data from a local JSON file."""
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    file_path = os.path.join(current_dir, "model_prices_and_context_window.json")
+    try:
+        with open(file_path, "r") as f:
+            data = json.load(f)
+        return data
+    except FileNotFoundError:
+        logger.error(f"Error: The file {file_path} was not found.")
+        return {}
+    except json.JSONDecodeError:
+        logger.error(f"Error: Could not decode JSON from the file {file_path}.")
+        return {}
+    except Exception as e:
+        logger.error(f"An unexpected error occurred: {e}")
+        return {}
+
+
+def litellm_pull_models_prices_and_context_window(
+    url: str = "https://raw.githubusercontent.com/BerriAI/litellm/refs/heads/main/model_prices_and_context_window.json",
```

Comment:
Pulls from URL (does not save to JSON, may want to save this to ~/.letta directory?)